### PR TITLE
Add back crypto.Signer support for ECDSA signing keys.

### LINKED ISSUE #226: Support crypto.Signer for signature generation
The documentation for `packet.PrivateKey` says the `PrivateKey` field may be:

```
// An *{rsa|dsa|elgamal|ecdh|ecdsa|ed25519}.PrivateKey or
// crypto.Signer/crypto.Decrypter (Decryptor RSA only).
```

But this is only really true for a PubKeyAlgoRSA type keys as of 6fef34b03b21e14c4dda47f3f8e5f9e3630a70ee (#111), which removed support for using a crypto.Signer for PubKeyAlgoECDSA.

Ideally all of the key types should support a crypto.Signer, so that keys which are not stored locally can be used to generate signatures. This is currently not possible.

Would there be any objection adding this support back for ECDSA (and possibly expanding the support for all of the key types)? Based on my reading, it should be relatively trivial to implement.

### PR DESCRIPTION
**Pull request by doryiii**
Currently only PubKeyAlgoRSA supports crypto.Signer. x/crypto/openpgp originally had support for ECDSA external crypto.Signer. This change adds that back.

This resolves #226 (at least for ECDSA signing keys).

### CODE REVIEW

**Comment by twiss** [comment]
Thanks! This looks reasonable to me, though the build fails because of a missing import of `big` :)

File: openpgp/packet/signature.go [hunk 1]
```diff
@@ -927,8 +929,17 @@ func (sig *Signature) Sign(h hash.Hash, priv *PrivateKey, config *Config) (err e
 			sig.DSASigS = new(encoding.MPI).SetBig(s)
 		}
 	case PubKeyAlgoECDSA:
-		sk := priv.PrivateKey.(*ecdsa.PrivateKey)
-		r, s, err := ecdsa.Sign(config.Random(), sk, digest)
+		var r, s *big.Int
+		var err error
```
Comment (twiss) [inline openpgp/packet/signature.go:933]: I think this is incorrect as it shadows the return parameter, hiding errors returned below
```suggestion
```

Comment (doryiii) [reply openpgp/packet/signature.go]: Fixed.

**Comment by doryiii** [comment]
Is there any documentation on how to run these regression tests locally so I can try and resolve them?

**Comment by twiss** [comment]
There was an issue on the main branch, should be fixed by https://github.com/ProtonMail/go-crypto/pull/229. So just rebasing should fix it :) Sorry about that.

**Review by twiss — APPROVED** [review]
Thanks!

**Pull request merged**